ehitchhiking-ui/src/app/shared/services/api.services/api.service.ts

```diff
@@ -1,30 +1,56 @@
+import {ResponseMessages} from './../../enums/response-message.enum';
 import {Injectable} from '@angular/core';
-import {HttpClient, HttpHeaders, HttpResponse} from '@angular/common/http';
+import {HttpClient, HttpHeaders, HttpResponse, HttpErrorResponse} from '@angular/common/http';
 import {Observable} from 'rxjs';
 import {CachingHttpParams} from '@shared/models/caching.http.params';
-import {RequestMethods} from '@shared/enums/request-enum';
+import {RequestMethods} from '@shared/enums/request-methods.enum';
 import {URL_REGISTRY} from '@shared/constants/urlRegistry';
+import {map, catchError} from 'rxjs/operators';
+import {NotificationService} from '../notification.service';
+import 'rxjs/add/operator/catch';
 
 @Injectable()
 export class ApiService {
 	static readonly apiUrl: string = 'http://localhost:4200/api/';
 
-	constructor(private http: HttpClient) {}
+	constructor(private http: HttpClient, private notificationService: NotificationService) {}
 
-	doGet(urlPath: string, isCacheable: boolean = false, parameters: any = null): Observable<HttpResponse<any>> | null {
-		return this.generateRequest(RequestMethods.GET, urlPath, parameters, isCacheable);
+	doGet(
+		urlPath: string,
+		isCacheable: boolean = false,
+		parameters: any = null,
+		isNotified: boolean = true
+	): Promise<any> {
+		return this.generateRequest(RequestMethods.GET, urlPath, isCacheable, isNotified, parameters);
+	}
+	doPost(urlPath: string, data: any = null, isNotified: boolean = true): Promise<any> {
+		return this.generateRequest(RequestMethods.POST, urlPath, false, isNotified, data);
 	}
-	doPost(urlPath: string, data: any = null): Observable<HttpResponse<any>> | null {
-		return this.generateRequest(RequestMethods.POST, urlPath, data, false);
+	doDelete(urlPath: string, parameters: any = null, isNotified: boolean = true): Promise<any> {
+		return this.generateRequest(RequestMethods.DEL, urlPath, false, isNotified, parameters);
 	}
-	doDelete(urlPath: string, parameters: any = null): Observable<HttpResponse<any>> | null {
-		return this.generateRequest(RequestMethods.DEL, urlPath, parameters, false);
+	getPut(urlPath: string, data: any = null, isNotified: boolean = true): Promise<any> {
+		return this.generateRequest(RequestMethods.PUT, urlPath, false, isNotified, data);
 	}
-	getPut(urlPath: string, data: any = null): Observable<HttpResponse<any>> | null {
-		return this.generateRequest(RequestMethods.PUT, urlPath, data, false);
+
+	doInitGet(): Promise<any> {
```

Comment:
what is a difference between doAuthGet and doInitGet?

Reply:
@DziubinaElena  spetific headers, different response (in init get spetific response from api), additional message in auth function